pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -220,11 +252,23 @@ func CreateGCECloud(projectID, zone, networkURL string, tokenSource oauth2.Token
 		networkURL = gceNetworkURL(projectID, networkName)
 	}
 
+	if len(managedZones) == 0 {
```

Comment:
Only in the case when multizone is set though, right?  This looks correct to me.

The weird structuring comes because when we are parsing the configuration, we don't yet have a GCE client, so we can't yet query the list of zones in the region.  So we pass nil to mean "all zones in region".